src/clr-angular/utils/drag-and-drop/if-dragged.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Directive, OnDestroy, TemplateRef, ViewContainerRef} from "@angular/core";
+import {Subscription} from "rxjs/Subscription";
+
+import {ClrDragEvent} from "./interfaces/drag-event";
+import {ClrDragEventListener} from "./providers/drag-event-listener";
+
+@Directive({selector: "[clrIfDragged]"})
+export class ClrIfDragged<T> implements OnDestroy {
+    private subscriptions: Subscription[] = [];
+    constructor(private template: TemplateRef<any>, private container: ViewContainerRef,
+                private dragEventListener: ClrDragEventListener<T>) {
```

Comment:
Could you please remind me where `ClrDragEventListener` is declared?

Reply:
So it looks like `clrIfDragged` has to be under `clrDraggable`. If thats the case, I think we need to warn the user only to use this directive under `clrDraggable`. If they don't we should throw an error. Check `dropdown-menu` for an example. It does something like this:

```
constructor(injector: Injector, @Optional() @Inject(POPOVER_HOST_ANCHOR) parentHost: ElementRef,
                @Optional() @SkipSelf() nested: ClrDropdownMenu) {
        if (!parentHost) {
            throw new Error("clr-dropdown-menu should only be used inside of a clr-dropdown");
        }
       super(...)
       ....
```

Do you think we need a check like this here too?

Reply:
I think we could/should have it. 🤔 Thanks for bringing this up!